.github/workflows/people.yml

```diff
@@ -71,40 +69,85 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: '18.x'
-          cache: 'yarn'
-          cache-dependency-path: src/frontend/yarn.lock
+
+      - name: Restore the frontend cache
+        uses: actions/cache@v3
+        id: cache-install
+        with:
+          path: src/frontend/
+          key: install-front-${{ hashFiles('src/frontend/**/yarn.lock') }}
 
       - name: Install dependencies
-        run: yarn install --frozen-lockfile
+        if: steps.cache-install.outputs.cache-hit != 'true'
+        run: cd src/frontend/ && yarn install --frozen-lockfile
+
+      - name: Cache install frontend
+        if: steps.cache-install.outputs.cache-hit != 'true'
+        uses: actions/cache@v3
+        with:
+          path: src/frontend/
+          key: install-front-${{ hashFiles('src/frontend/**/yarn.lock') }}
+
+  build-front:
+    runs-on: ubuntu-latest
+    needs: install-front
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Restore the frontend cache
+        uses: actions/cache@v3
+        id: cache-install
+        with:
+          path: src/frontend/
+          key: install-front-${{ hashFiles('src/frontend/**/yarn.lock') }}
+
+      - name: Build CI App
+        run: cd src/frontend/ && yarn ci:build
+
+      - name: Cache build frontend
+        uses: actions/cache@v3
+        with:
+          path: src/frontend/
+          key: build-front-${{ github.run_id }}
+        
+  test-front:
+    runs-on: ubuntu-latest
+    needs: install-front
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Restore the frontend cache
+        uses: actions/cache@v3
+        id: cache-install
+        with:
+          path: src/frontend/
+          key: install-front-${{ hashFiles('src/frontend/**/yarn.lock') }}
 
       - name: Test App
-        run: yarn app:test
+        run: cd src/frontend/ && yarn app:test
 
   lint-front:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: src/frontend/
```

Comment:
A single question, why don't all frontend jobs are running with this default working directory?
```
    defaults:
      run:
        working-directory: src/frontend/
```

Reply:
It is to avoid confusion, by example here, if you don't specify the path, even with a working directory, it will fail:
https://github.com/numerique-gouv/people/blob/415b86bb97a6c6efd8169819a3f40cbf78570123/.github/workflows/people.yml#L142
So I put everythink with the path, it is a bit redundant, but easy to read and understand.